modules/core/04-channel/types/msgs.go

```diff
@@ -499,18 +499,57 @@ var _ sdk.Msg = &MsgChannelUpgradeInit{}
 
 // NewMsgChannelUpgradeInit constructs a new MsgChannelUpgradeInit
 // nolint:interfacer
-func NewMsgChannelUpgradeInit() *MsgChannelUpgradeInit {
-	return &MsgChannelUpgradeInit{}
+func NewMsgChannelUpgradeInit(
+	portID, channelID string,
+	proposedUpgradeChannel Channel,
+	counterpartyTimeoutHeight clienttypes.Height,
+	counterpartyTimeoutTimestamp uint64,
+	signer string,
+) *MsgChannelUpgradeInit {
+	return &MsgChannelUpgradeInit{
+		PortId:                       portID,
+		ChannelId:                    channelID,
+		ProposedUpgradeChannel:       proposedUpgradeChannel,
+		CounterpartyTimeoutHeight:    counterpartyTimeoutHeight,
+		CounterpartyTimeoutTimestamp: counterpartyTimeoutTimestamp,
+		Signer:                       signer,
+	}
 }
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgChannelUpgradeInit) ValidateBasic() error {
-	return errors.New("error method not implemented")
+	if err := host.PortIdentifierValidator(msg.PortId); err != nil {
+		return sdkerrors.Wrap(err, "invalid port ID")
+	}
+
+	if !IsValidChannelID(msg.ChannelId) {
+		return ErrInvalidChannelIdentifier
+	}
+	if msg.ProposedUpgradeChannel.State != INITUPGRADE {
+		return sdkerrors.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.ProposedUpgradeChannel.State)
+	}
+	if msg.CounterpartyTimeoutHeight.IsZero() {
```

Comment:
do both timeoutheight and timeout timestamp need to be set?

Reply:
I see in the spec in `chanOpenInit` that at least one of them should be set:

```golang
// either timeout height or timestamp must be non-zero
abortTransactionUnless(counterpartyTimeoutHeight != 0 || counterpartyTimeoutTimestamp != 0)
```

Reply:
cc @AdityaSripal

Reply:
Yes at least one needs to be set

Reply:
Same as `MsgChannelUpgradeTry` PR - https://github.com/cosmos/ibc-go/pull/1836#discussion_r935402515